Plutarch/List.hs

```diff
@@ -276,10 +305,14 @@ pzip ::
   Term s (list a :--> list b :--> list (PPair a b))
 pzip = phoistAcyclic $ pzipWith' $ \x y -> pcon (PPair x y)
 
--- Horribly inefficient.
-plistEquals :: (PIsListLike list a, PElemConstraint list PBool, PEq a) => Term s (list a :--> list a :--> PBool)
+-- | / O(min(n, m)) /. Check if two lists are equal.
+plistEquals :: (PIsListLike list a, PEq a) => Term s (list a :--> list a :--> PBool)
```

Comment:
Perhaps we can give `PBuiltinList` and a `PEq` instance now that `plistEquals` is fairly efficient,

```
-- Bench results against `Plutarch.List.plistEquals`
> scriptSize $ compile $ plistEquals @PBuiltinList @PInteger
184
> scriptSize $ compile $ plistEquals' @PBuiltinList @PInteger
91
> evalT $ plistEquals @PBuiltinList @PInteger # pconstant [1, 2, 3] # pconstant [1, 2, 3]
Right (ExBudget {exBudgetCPU = ExCPU 25843624, exBudgetMemory = ExMemory 59392},[],Program () (Version () 1 0 0) (Constant () (Some (ValueOf bool True))))
> evalT $ plistEquals' @PBuiltinList @PInteger # pconstant [1, 2, 3] # pconstant [1, 2, 3]
Right (ExBudget {exBudgetCPU = ExCPU 9398080, exBudgetMemory = ExMemory 20846},[],Program () (Version () 1 0 0) (Constant () (Some (ValueOf bool True))))
> evalT $ plistEquals @PBuiltinList @PInteger # pconstant [1, 2, 3, 4] # pconstant [1, 2, 3]
Right (ExBudget {exBudgetCPU = ExCPU 12939651, exBudgetMemory = ExMemory 30928},[],Program () (Version () 1 0 0) (Constant () (Some (ValueOf bool False))))
> evalT $ plistEquals' @PBuiltinList @PInteger # pconstant [1, 2, 3, 4] # pconstant [1, 2, 3]
Right (ExBudget {exBudgetCPU = ExCPU 9398080, exBudgetMemory = ExMemory 20846},[],Program () (Version () 1 0 0) (Constant () (Some (ValueOf bool False))))
> evalT $ plistEquals @PBuiltinList @PInteger # pconstant [] # pconstant [1, 2, 3]
Right (ExBudget {exBudgetCPU = ExCPU 7497147, exBudgetMemory = ExMemory 19064},[],Program () (Version () 1 0 0) (Constant () (Some (ValueOf bool False))))
> evalT $ plistEquals' @PBuiltinList @PInteger # pconstant [] # pconstant [1, 2, 3]
Right (ExBudget {exBudgetCPU = ExCPU 1639885, exBudgetMemory = ExMemory 4664},[],Program () (Version () 1 0 0) (Constant () (Some (ValueOf bool False))))
```

Reply:
sounds good to me

Reply:
nvm, `PBuiltinList` already has a `PEq` instance. That works out very well.